### ISSUE TITLE
External Display receives no signal anymore

### ISSUE THREAD

**Issue by Joghurtbecher08115** [dortania/OpenCore-Legacy-Patcher#190]
Hello,

I run this patch on my late 2012 21.5' iMac with now macOS Big Sur 11.3 and everything works great so far except the external display (Acer G226HQL 1080p monitor) which ist connected to the iMac via Thunderbolt -> HDMI cable. The monitor worked fine before under Catalina and still works with my 2017 Macbook Pro under Big Sur, however with my iMac it never gets a signal.

It seems similar to this issue: https://github.com/dortania/OpenCore-Legacy-Patcher/issues/157

However what ist strange is: If I connect my Macbook Pro to the display which receives a signal and unplug it and immediately connect the iMac the monitor gets a signal from the iMac. So I have a workaround for now, but I need my iMac and the second screen these days every day for work, so I would prefer being able to use my external display with the step with the Macbook ;-)

If you need any additional information let me know and I will see if I can help, however not an expert yet when it comes to patching macOS like this :-)

**Comment by khronokernel** [comment]
Hello, would it be possible to send us IORegs of the following?

* Catalina with display plugged in
* Big Sur with display plugged in
* Big Sur with display disconnected

Gives us a better idea as to where to look. For the IOReg, simply run the below app and run File -> SaveAs:
https://github.com/khronokernel/IORegistryClone/blob/master/ioreg-210.zip?raw=true

Also if you're able to, can you upload your OpenCore EFI OCLP built you? It'll be on the EFI Partition OCLP installed to, running MountEFI is the easiest way to mount it:
https://github.com/corpnewt/MountEFI

**Comment by Joghurtbecher08115** [comment]
Hello, thanks for the quick response, I just ran the IORegistryExplorer application and created three files, one with the monitor not plugged in, one with the monitor plugged in and receiving no signal and one with the monitor being plugged and and having a signal (after doing the Macbook workaround mentioned above). Right now I have no Catalina installation anymore, if really necessary I will see what I can do the coming weekend and install it on an external drive.

I downloaded the MountEFI tool and ran ist and it gives me several options in the Terminal, could you give me a some instructions what I should to in the Terminal?

Thanks again and greetings!

[Archiv.zip](https://github.com/dortania/OpenCore-Legacy-Patcher/files/6423650/Archiv.zip)

**Comment by paltrowlalic** [comment]
Hi, I believe your problem is very similar to mine (No. 157). To use the Thunderbolt display I have to
1. Reinstall Catalina 10.15.7 using dosdude1's Catalina patcher
2. Install some post patches (see the image attached) then the Thunderbolt display will be detected.
![image](https://user-images.githubusercontent.com/19687506/117359553-788dc980-aeaf-11eb-9408-f30b77e29b97.png)

One difference is that the machine is recognized as iMac12,2 in Catalina but differently in BigSur using OPLC (MacPro15,1 or iMac15,1 I can't remember). My guess is that there is some conflict between the board-id and some display-related patches. Hope this helps!

**Comment by Ausdauersportler** [comment]
This patch set for iMac 2011 includes (invisible) the AppleGraphicsControl patches to enable the display ports and avoid a black screen on boot with NVIDIA dGPU. If this would be the solution there is a more easy OpenCore way (agdpmod=vit9696). But I doubt it will be that easy.

**Comment by paltrowlalic** [comment]
Hi @Ausdauersportler thanks for the reply. Do you mean change the key value to vit9696 in the config.plist file?
I will give it a try during weekend!
![image](https://user-images.githubusercontent.com/19687506/117366674-0b7f3180-aeb9-11eb-8bd2-00adc410404c.png)

**Comment by khronokernel** [comment]
@Ausdauersportler Might be actually, we set `agdpmod=pikera` however from the looks of things it seems that's incorrect. Would appreciate if @Joghurtbecher08115 and @paltrowlalic could test that out

TL;DR is to mount the drive you installed OpenCore on with [mountEFI](https://github.com/corpnewt/MountEFI), look inside EFI/OC and see the config.plist file. Open it and find the `PciRoot(0x0)/Pci(0x1,0x0)/Pci(0x0,0x0)` entry under DeviceProperties. Here you'll want to edit from `agdpmod | string | pikera` to `agdpmod | string | vit9696`

**Comment by paltrowlalic** [comment]
I will try tomorrow evening and let you guys know the resutls.

**Comment by paltrowlalic** [comment]
Hi all, I would like to let you know that it worked! Thank you for the help!!
![image](https://user-images.githubusercontent.com/19687506/117435259-90a03000-af25-11eb-9581-70870c6c0218.png)
![image](https://user-images.githubusercontent.com/19687506/117435334-a7468700-af25-11eb-9da5-f726a4e9d6dd.png)
